drivers/retention/Kconfig

```diff
@@ -0,0 +1,28 @@
+# Copyright (c) Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig RETENTION
+	bool "Data retention support"
+	help
+	  Enables support for retaining data which can survive a reboot (and
```

Comment:
I think this should be expanded to explain how these devices different from FLASH.

Reply:
Added